### ISSUE TITLE
Extract common.MapStr to its own package

### ISSUE THREAD

**Issue by ruflin** [elastic/beats#14734]
If someone wants to use `common.MapStr` outside Beats today, it means including lots of other dependency because of the `common` package contains many other othings. It would be nice if `MapStr` could end up in its own package without other dependencies.

This is coming out of https://github.com/elastic/package-registry/pull/157

**Comment by mtojek** [comment]
I think it's the right moment to extract it. There are more places that can leverage from that:

https://github.com/elastic/integrations/blob/master/dev/import-beats/common.go
https://github.com/elastic/elastic-package/blob/master/internal/common/mapstr.go
https://github.com/elastic/package-spec/blob/master/code/go/internal/validator/mapstr.go

Do you have any proposals for the package name? `github.com/elastic/mapstr` ? 

/cc @urso @ycombinator

**Comment by urso** [comment]
If possible I'd rather get rid of MapStr, instead of depending on it more and more. If possible, prefer proper types. Is there a good reason why MapStr was prefered in these other projects?

Anyways, the ask seems to be about being able to reuse MapStr outside of Beats. A separate repository would make sense, such that not all projects depend on Beats.

**Comment by mtojek** [comment]
> If possible I'd rather get rid of MapStr, instead of depending on it more and more. If possible, prefer proper types. Is there a good reason why MapStr was prefered in these other projects?

Proper typing might hard to introduce in the following cases:

1. Processing Kibana dashboards. I'm not sure if there is schema defined somewhere and it would be hard to follow potential changes (see: https://github.com/elastic/elastic-package/blob/master/internal/builder/dashboards.go#L58).
2. Expanding different manifest YAML files for validation against JSON schema. JSON schema depends only on expanded fields (see: https://github.com/elastic/package-spec/blob/master/code/go/internal/validator/folder_item_content.go#L45).
